MatrixKit/Models/Account/MXKAccountManager.m

```diff
@@ -493,37 +487,64 @@ - (BOOL)isPushAvailable
 
 #pragma mark -
 
-- (void)loadAccounts
+// Return the path of the file containing stored MXAccounts array
+- (NSString*)accountFile
 {
-    NSUserDefaults *sharedDefaults = [MXKAppSettings standardAppSettings].sharedUserDefaults;
+    NSString *matrixKitCacheFolder = [MXKAppSettings cacheFolder];
+    return [matrixKitCacheFolder stringByAppendingPathComponent:kMXKAccountsKey];
+}
 
-    NSData *accountData = [sharedDefaults objectForKey:kMXKAccountsKey];
+- (void)loadAccounts
+{
+    NSLog(@"[MXKAccountManager] loadAccounts");
 
-    if (!accountData)
+    NSString *accountFile = [self accountFile];
+    if ([[NSFileManager defaultManager] fileExistsAtPath:accountFile])
     {
-        // Migration of accountData from [NSUserDefaults standardUserDefaults] to sharedDefaults (shared between apps and extensions)
-        NSData *oldAccountData = [[NSUserDefaults standardUserDefaults] objectForKey:kMXKAccountsKey];
-        if (oldAccountData)
+        NSDate *startDate = [NSDate date];
+        mxAccounts = [NSKeyedUnarchiver unarchiveObjectWithFile:accountFile];
+        NSLog(@"[MXKAccountManager] loadAccounts. %tu accounts loaded in %.0fms", mxAccounts.count, [[NSDate date] timeIntervalSinceDate:startDate] * 1000);
+    }
+    else
+    {
+        // Migration of accountData from [NSUserDefaults standardUserDefaults] to a file
+        NSUserDefaults *sharedDefaults = [MXKAppSettings standardAppSettings].sharedUserDefaults;
+
+        NSData *accountData = [sharedDefaults objectForKey:kMXKAccountsKey];
+        if (!accountData)
         {
-            [sharedDefaults setObject:oldAccountData forKey:kMXKAccountsKey];
-            [sharedDefaults synchronize];
+            // Migration of accountData from [NSUserDefaults standardUserDefaults] to sharedDefaults (shared between apps and extensions)
+            NSData *oldAccountData = [[NSUserDefaults standardUserDefaults] objectForKey:kMXKAccountsKey];
+            if (oldAccountData)
+            {
+                [sharedDefaults setObject:oldAccountData forKey:kMXKAccountsKey];
+                [sharedDefaults synchronize];;
 
-            // TODO: Erase old location of accountData when the app goes to the app store
-            //[[NSUserDefaults standardUserDefaults] removeObjectForKey:kMXKAccountsKey];
-            //[[NSUserDefaults standardUserDefaults] synchronize];
+                NSLog(@"[MXKAccountManager] loadAccounts: performed data migration #1");
```

Comment:
Why do you store here this old account in shared Defaults?
I would only check old account storage if none is available in shared defaults:
```
if (!accountData)
{
   accountData = [[NSUserDefaults standardUserDefaults] objectForKey:kMXKAccountsKey];
}
```